presto-main/src/main/java/io/prestosql/server/ExecutingStatementResourceConfig.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.DataSize;
+import io.airlift.units.MaxDataSize;
+import io.airlift.units.MinDataSize;
+
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+
+public class ExecutingStatementResourceConfig
+{
+    private DataSize serverTargetResultSize = DataSize.of(1, MEGABYTE);
+
+    @MinDataSize("1MB")
+    @MaxDataSize("128MB")
+    public DataSize getServerTargetResultSize()
+    {
+        return serverTargetResultSize;
+    }
+
+    @Config("query.target-result-size")
```

Comment:
I think parameter name does not capture the meaning very well. It matches internal parameter name but IMO it is not very much understandable to external user. Maybe sth like `query.transport-chunk-size` would be better?

Reply:
I think `fetch-size` makes sense since everyone using JDBC has an idea of what it is.

The only downside is fetch-size is generally a number of rows rather than data size.

Reply:
I think this is actually a considerable downside. fetch size of 1 makes sense (one row at a time.
target-result-size of 1 (1 what? byte?) does not make sense

also, we could make it more clear that it applies to the client side. 
not sure whether `query.` suffices

Reply:
I am open to change it to whatever value is appropriate. I like `query.transport-chunk-size` though.

Reply:
How about `query.results_transfer_rate` ?